### ISSUE TITLE
Video File Playback via FFMPEG core?

### ISSUE THREAD

**Issue by metaColin** [madcock/sf2000_multicore_cores#2]
I've been enjoying the heck out of Multicore.

I know Retroarch has an FFMPEG core and it got me thinking that I could put some classic Saturday morning cartoons in a folder and enjoy them on my SF2000.

Would this be possible with the current FFPEG core that is in this repo? If yes, what would the procedure be to add videos? Is there particular encoding likely to work best?

If not, what would that require?

**Comment by madcock** [comment]
Glad to hear you're enjoying Multicore! :)

Unfortunately, not all of the cores I've forked here in github actually build (yet) for the SF2000.  And the FFMPEG core is one of those. It seems to require thread functions that we don't have on the platform, at least with the current environment that reuses the existing firmware. Previous to multicore we were using a different OS build (hcrtos) and it supported some of the things we're now missing. It's possible we'll change back to that at come point, or find a way to add some of the missing functionality, so I've kept these non-working cores around.

I've got a [spreadsheet here](https://docs.google.com/spreadsheets/d/1BDPqLwRcY2cN7tObuyW7RzLw8oGyY9XGLS1D4jLgz2Q/edit?usp=sharing) that lists all the cores (most of which are in github, though some that don't build have not been added yet). I try to keep it up to date in terms of status for everything, and also add a few little notes to indicate what is going on with each core. In general, green stuff works, yellow stuff works but has issues (some which may not be fixable like performance issues, etc.), orange builds but doesn't so anything or much of anything at all but has the potential to be fixed at working once we figure it out, red doesn't build and may never work (except perhaps with lots of effort), purple has unsupported build environments but otherwise is unknown so some of these might end up working with some effort, and dark purple actually work as well as they do on other Retroarch platforms but are buggy and unfinished so haven't been released.

The only way I know, right now, to play videos under SF2000 multicore would be to pack them into a custom GBA ROM and play them that way. I haven't done it myself, but I know others on the discord have used GBA homebrew movie playing tools to do that. It's cumbersome, and it would reencode the video to be the supported resolution/colors of the GBA, so it might not look great, but it does work. But it's not like copying a mp4 file into a specific place. :( For instance, there's a [Rick Roll GBA cartridge](https://github.com/shinyquagsire23/GBA-Rick).

I think this is what some people have been using to do it: [Meteo](https://www.gamebrew.org/wiki/Meteo_Avi-2-GBA)

We don't really have a good audio playing solution either, although the karaoke core (Pocket CDG) can play MP3 files if they have a matching CDG file.
